app/domain/operations/hbx_enrollments/drop_enrollment_members.rb

```diff
@@ -11,51 +11,101 @@ class DropEnrollmentMembers
+        custom_params = custom_params(params[:hbx_enrollment], new_effective_date)
+        clone_enrollment = Operations::HbxEnrollments::Clone.new.call({hbx_enrollment: params[:hbx_enrollment], effective_on: new_effective_date, options: custom_params})
+        return clone_enrollment if clone_enrollment.failure?
 
-        reinstatement.force_select_coverage! if reinstatement.may_reinstate_coverage?
-        reinstatement.begin_coverage! if reinstatement.may_begin_coverage? && reinstatement.effective_on <= TimeKeeper.date_of_record
+        new_enrollment = clone_enrollment.value!
+        new_enrollment.hbx_enrollment_members.delete_if {|mem| non_eligible_members.pluck(:applicant_id).include?(mem.applicant_id)}
+        set_product_id(params[:hbx_enrollment], new_enrollment)
+        new_enrollment.check_for_subscriber
+        new_enrollment.generate_hbx_signature
+        new_enrollment.select_coverage! if new_enrollment.may_select_coverage?
+        update_member_effective_dates(new_enrollment)
+        return Failure('Failed to save dropped member enrollment') unless new_enrollment.persisted?
 
-        params[:hbx_enrollment].update_attributes!(terminated_on: terminate_date)
+        get_household_applied_aptc(new_enrollment, new_effective_date) if params[:hbx_enrollment].applied_aptc_amount > 0
+        terminate_base_enrollment(params[:hbx_enrollment], new_enrollment, termination_date)
 
         dropped_member_info = []
         dropped_enr_members.each do |member_id|
           member = params[:hbx_enrollment].hbx_enrollment_members.where(id: member_id).first
           dropped_member_info << {hbx_id: member.hbx_id,
                                   full_name: member.person.full_name,
-                                  terminated_on: terminate_date}
+                                  terminated_on: termination_date}
         end
 
         Success(dropped_member_info)
       end
 
+      def update_member_effective_dates(new_enrollment)
+        new_enrollment.hbx_enrollment_members.each do |member|
+          member.update_attributes(eligibility_date: new_enrollment.effective_on, coverage_start_on: new_enrollment.effective_on)
+        end
+      end
+
+      def terminate_base_enrollment(base_enrollment, new_enrollment, termination_date)
+        if base_enrollment.may_terminate_coverage? && (new_enrollment.effective_on > base_enrollment.effective_on)
+          base_enrollment.terminate_coverage!
+          base_enrollment.update_attributes!(terminated_on: termination_date)
+        elsif base_enrollment.may_cancel_coverage?
+          base_enrollment.cancel_coverage!
+        end
+      end
+
+      def custom_params(base_enrollment, new_effective_date)
+        { predecessor_enrollment_id: base_enrollment.id,
+          rating_area_id: ::BenefitMarkets::Locations::RatingArea.rating_area_for(base_enrollment.consumer_role.rating_address, during: new_effective_date)&.id }
+      end
```

Comment:
Above logic sets rating area to nil if not found.

Should return failure if unable to find rating area, setting rating area to nil results in 0$ or 1$ premiums.